### PR TITLE
monitoring: improved coverage (PROJQUAY-3290)

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Improved the coverage for Deployment health checks. We now take into
account possible overrides.

### CODE REVIEW

File: pkg/cmpstatus/clair.go [hunk 1]
```diff
@@ -73,6 +73,30 @@ func (c *Clair) Check(ctx context.Context, reg qv1.QuayRegistry) (qv1.Condition,
...
+	// users are able to override the number of replicas. if they do override it to zero
+	// we expect zero replicas to be running.
+	replicas := qv1.GetReplicasOverrideForComponent(&reg, qv1.ComponentClair)
+	scaleddown := replicas != nil && *replicas == 0
+	if scaleddown {
+		if dep.Status.AvailableReplicas == 0 {
+			return qv1.Condition{
+				Type:           qv1.ComponentClairReady,
+				Reason:         qv1.ConditionReasonComponentReady,
+				Status:         metav1.ConditionTrue,
```
Comment (dmage) [inline pkg/cmpstatus/clair.go:85]: Base reports itself as Ready: false when it's scaled down, but Clair as Ready: true. Is it intentionally?

File: pkg/cmpstatus/postgres.go [hunk 2]
```diff
@@ -41,6 +41,11 @@ func (p *Postgres) Check(ctx context.Context, reg qv1.QuayRegistry) (qv1.Conditi
 	}
 
 	nsn := types.NamespacedName{
+		Namespace: reg.Namespace,
+		Name:      fmt.Sprintf("%s-quay-app-upgrade", reg.Name),
+	}
+
+	nsn = types.NamespacedName{
```
Comment (dmage) [inline pkg/cmpstatus/postgres.go]: What's going on here?..

File: pkg/cmpstatus/clair.go (hunk 1 above)
Comment (dmage) [reply pkg/cmpstatus/clair.go:85]: nvmd, I misread changes in pkg/cmpstatus/base.go.

File: pkg/cmpstatus/postgres.go (hunk 2 above)
Comment (ricardomaraschini) [reply pkg/cmpstatus/postgres.go]: I don't know, I don't know! Fixed.

**Review by dmage — APPROVED** [review]

**Pull request merged**